puppet/lib/puppet/indirector/facts/puppetdb.rb

```diff
@@ -30,12 +33,12 @@ def save(request)
           # when we attempt to use ActiveSupport 2.3.16 on RHEL 5 with
           # legacy storeconfigs.
           "environment" => request.options[:environment] || request.environment.to_s,
-          "producer_timestamp" => request.options[:producer_timestamp] || Time.now.iso8601(5),
+          "producer_timestamp" => Puppet::Util::Puppetdb.to_wire_time(current_time),
           "producer" => Puppet[:node_name_value]
         }
       end
 
-      submit_command(request.key, payload, CommandReplaceFacts, 5)
+      submit_command(request.key, payload, CommandReplaceFacts, 5, current_time.utc)
```

Comment:
The distinction between `.utc` here, and without that above, is interesting. Should submit command always be dealing with utc times? If so, should it invoke `.utc` itself?